### ISSUE TITLE
revise submit endpoint

### ISSUE THREAD

**Issue by pbelmann** [jkrue/jobproxy#60]
While writing tests for #36 I noticed a few possible improvements regarding the submit endpoint:

**Ports**

* The container object has a list of ports objects. The mounts object contains a port object. I suggest that the container has a list of port objects. So we do not need the ports object.

* The fields host and container should be marked as NotEmpty and NotNull

**Mounts**

* The container object has a list of mounts objects. The mounts object contains a mount object. I suggest that the container has a list of mount objects. So we do not need the mounts object.

**Mount**

* Modi: At the moment the rest endpoint accepts any string. We should restrict it to the allowed values (rw and ro)

* all fields should be marked as NotNull
 
* host path is allowed to be null since, a volume can be specified by using container

**Additional Points**:

* add container cmd field as mentioned by @jkrue  in #56 .

* clarify whether stderr and stdout are path or not as mentioned in #58

**Comment by jkrue** [comment]
I agree with everything. The nested mounts/mount ports/port object combination seems to be a result from the xjc code generation process. However as I see the necessary changes  affects _only_ the Java code but not the REST API. 

I understand _stdout_ and _stderr_ everytime as (directory) path. Well, it may be framework dependent wether it points to  an directory or file. @pbelmann, we have to discuss this

**Comment by pbelmann** [comment]
* All optional fields should have a method like hasContainer for container field.
